pkg/cmd/cli/cmd/rsync/util.go

```diff
@@ -118,3 +121,32 @@ func rsyncSpecificFlags(o *RsyncOptions) []string {
 	}
 	return flags
 }
+
+type podApiChecker struct {
+	factory   *clientcmd.Factory
+	namespace string
+	podNames  []string
+}
+
+// CheckPods will check if pods exists in the provided context
+func (p podApiChecker) CheckPods() error {
+	mapper, typer := p.factory.Object()
```

Comment:
why not just get the client from f
and do client.Pods(p.namespace).Get(podName)?